src/apphosting/secrets/dialogs.ts

```diff
@@ -0,0 +1,148 @@
+import * as clc from "colorette";
+const Table = require("cli-table");
+
+import * as apphosting from "../../gcp/apphosting";
+import * as prompt from "../../prompt";
+import * as iam from "../../gcp/iam";
+import * as utils from "../../utils";
+import { logger } from "../../logger";
+
+interface Metadata {
+  location: string;
+  id: string;
+  accounts: string[];
+}
+
+export function serviceAccountsForBackend(
+  projectNumber: string,
+  backend: apphosting.Backend,
+): string[] {
+  if (backend.serviceAccount) {
+    return [backend.serviceAccount];
+  }
+  return [
+    iam.getDefaultCloudBuildServiceAgent(projectNumber),
+    iam.getDefaultComputeEngineServiceAgent(projectNumber),
+  ];
+}
+
+export function toMetadata(projectNumber: string, backends: apphosting.Backend[]): Metadata[] {
+  const metadata: Metadata[] = [];
+  for (const backend of backends) {
+    const [, , , location, , id] = backend.name.split("/");
```

Comment:
Let's include a line comment with an example of a raw backend.name value

Reply:
Done